client/controllers/public/channel.js

```diff
@@ -0,0 +1,9 @@
+Template.channel.onCreated(function() {
+  Meteor.subscribe('oneChannel', FlowRouter.getParam('id'));
```

Comment:
This should be wrapped in an autorun block like the [code example here](https://kadira.io/academy/meteor-routing-guide/content/subscriptions-and-data-management/with-blaze). 

That way if the id parameter changes, the subscription will be automatically stopped.